### ISSUE TITLE
Non-ADB observation broken on FireOS 5.0.5.1

### ISSUE THREAD

**Issue by RoxKilly** [sphinx02/FireStarter#56]
FireTV 2
FireOS 5.0.5.1
FireStarter 3.2.3

Non-ADB observation doesn't work, even from the Amazon home screen. From Amazon home, pressing the home button again doesn't do anything.  ADB-based observation works (Amazon screen flashes for a split second, then Kodi opens).  This happened immediately after I upgraded from FireOS 5.0.5 to 5.0.5.1

To replicate:
1.  Go into FireStarter settings
2.  In `Background Observation` section, check `Enable background observation`, check `Fallback to Non-ADB Observation` and remove/uncheck `Home-Detection via ADB`
3.  Go to the Amazon home screen, and press the `home` button.  Nothing happens

The following makes no difference since this issue is about Non-ADB, but I made sure to add back the `adb` program that Amazon removed from FireOS 5.0.5.1 ([guide to add back adb](http://www.aftvnews.com/how-to-restore-internal-adb-client-capabilities-on-a-rooted-fire-tv/))

**Comment by infinityhaitian** [comment]
I have that same problem now, what do I do now?

**Comment by jvanda1** [comment]
....is llama still an option? Or did that get blacklisted also? 
Maybe llama could be used to shortcut/open fire starter...?

**Comment by infinityhaitian** [comment]
Please Fix firestarter.. I have donated

**Comment by TarkusLV** [comment]
Llama is no longer necessary, because you can now open Kodi direct from the Fire TV home screen.  You can add sideloaded apps to your Favorites and they'll show up in your Recent list.

There is a fork of FireStarter called FireInstaller that handles only the installing/updating portion for Kodi, and it works great.  That is really all I need out of FireStarter, now that I can load Kodi right from my default Home menu.

**Comment by meliton** [comment]
I know 5.2.1.0 broke the Home button presses, so... how is 5.2.1.0 tracking Home button presses? Is it possible to piggyback on their button detection?

**Comment by meliton** [comment]
@sphinx02, I noticed that with 5.2.1.0, a Home Button press will trigger these events in the log

```
W/TelecomManager(  987): Telecom Service not found.
D/CustomHomeButtonHandler(  987): getCustomHomeReceiver: package: 'com.amazon.tv.launcher'
D/CustomHomeButtonHandler(  987): doesPackageHandleHome: 'com.amazon.tv.launcher' = false
I/ActivityManager(  987): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.HOME] flg=0x10200000 cmp=com.amazon.tv.launcher/.ui.HomeActivity} from uid 1000 on display 0
I/ActivityManager(  987): START u0 {act=android.intent.action.VIEW cat=[com.amazon.device.intent.category.LAUNCHER_MENU] flg=0x14000000 cmp=com.amazon.tv.launcher/.ui.HomeActivity (has extras)} from uid 32071 on display 0
```

Maybe when the `CustomHomeButtonHandler` event is seen, AppStarter can come back to the front.

**Comment by esc0rtd3w** [comment]
using meliton's suggestion and looking at line 518 in "BackgroundHomeButtonObserverThreadADB.java", which was removed for some reason (assuming integration into AppStarter)

 if(message.startsWith("I/ActivityManager") && message.contains("act=android.intent.action.MAIN cat=[android.intent.category.HOME]"))

can be replaced with 

 if(message.startsWith("I/ActivityManager") && message.contains("act=android.intent.action.VIEW cat=[com.amazon.device.intent.category.LAUNCHER_MENU]"))

or

 if(message.startsWith("D/CustomHomeButtonHandler") && message.contains("doesPackageHandleHome"))

**Comment by RoxKilly** [comment]
@esc0rtd3w  Is this something you've tested and can confirm to work, or are you brainstorming?

**Comment by esc0rtd3w** [comment]
brainstorming....for 5.0.5.1. I have since been rooting and downgrading all my firesticks to 5.0.5 and using FireStopper, so this app is mainly useless to me personally.

**Comment by RoxKilly** [comment]
At first I downgraded but I wasn't happy with falling further and further behind with each new firmware release.  I've since upgraded to [5.2.1.0](http://forum.xda-developers.com/fire-tv/development/firetv-2-sloane-prerooted-stock-images-t3309783) and disabled this app from my device.  I'm using [rbox FireTV mods](http://forum.xda-developers.com/showthread.php?t=2786618) which will allow an alternate launcher and will bind the home button to that launcher.  That along with [XBMC Launcher](http://apps.evozi.com/apk-downloader/?id=se.blunden.xbmclauncher), which makes Kodi that launcher.

Now I boot into Kodi. And when I press the home button, Kodi opens. And I have the latest firmware.  Works for me.

PS: rbox' FireTV mods requires that you have the [XPosed framework](http://www.aftvnews.com/how-to-install-xposed-framework-on-the-fire-tv-1-or-2-running-fire-os-5/), which requires root.

**Comment by esc0rtd3w** [comment]
I like your XBMC Launcher suggestion. For me, booting directly into Kodi is an issue i struggle with for the firestick...lol. It has its ups and downs. I will eventually resort to that and integrate the Android App shortcut stuff into my Kodi build XML probably. 

I do like staying on 5.0.5 because i disable all the Amazon bloat (while retaing Settings only) and unroot to give to customers. I disable Help and Parental Controls as well. I also disable the Factory Reset button and Amazon OTA update services and activities. I get an average extra 150+MB RAM free (Kodi cold boots as high as 270+MB free) after doing this. I am still in the process of building a clean Amazon-Free system partition without settings problems. 

I was trying to get the stock 5.1 Lollipop Settings.apk (being on FireOS 5.0.5) working, but could not successfully resign the APK!!

**Comment by bigyin82** [comment]
Roxkilly, I used the home button launcher on the old version of firestarter and the temp firestopper  to launch kodi because I don't like to have the stick registered to an account. Would rbox fire tv mods help me to do this and unregister the stick. Thanks.

**Comment by RoxKilly** [comment]
@esc0rtd3w Have you not gotten any complaints from customers about not being able to update?

@bigyin82 I'm afraid I know nothing about the stick so I can't answer your question. I didn't even know you could use these devices without registering them. I would have liked to do that myself. I don't like the idea of these companies collecting so much info about me from different sources.

**Comment by bigyin82** [comment]
Yeah the home button option meant you didn't have to register the stick so it was ideal. Thanks for the response.

**Comment by esc0rtd3w** [comment]
@RoxKilly I have not had any complaints, as most customers do not even understand what the firestick is. An unexpected Amazon update constantly disables and kills things we love....so Amazon is gone!

For anyone interested, I keep a GitHub for my Firestick Loader here: https://github.com/esc0rtd3w/firestick-loader

Currently, the only issue i have encountered is the Amazon Android Remote App does not connect (assuming from a disabled service), which I will probably test this weekend to find the culprit activity/service. and re-enable to fix that issue.

UPDATE: ALL ISSUES FIXED! BUILD IS STABLE

**Comment by jamesfrom910** [comment]
im not able to bind home button at all on the latest fireos version. adb observation fails after switching it on and off again, attempting without no matter where i am the home button acts natively.  also when i try to update the app it says cant locate download url. that doesnt happen with spmc/kodi though.  pretty sure the fault is the user not the program and ill figure it out,  fire os 5.2.4.0, btw ty for this launcher. love the ability to edit wallpaper.
